### DISCUSSION TITLE
Config Datei suchen & bearbeiten

### DISCUSSION THREAD

**Discussion by RasputinBerry** [flobz/psa_car_controller#812] [Q&A]
Ich nutze den PSACC (testweise) in der QNAP Container Station _(... und kenne mich (noch) nicht mit der App bzw. dem Docker wirklich gut aus ...)_

WO finde ich dieses config-file um die API's WIE einzutragen? 🤔

![PSA](https://github.com/flobz/psa_car_controller/assets/56359831/645073ec-fe00-4130-adbd-fb5bc89c63e2)

Danke für die Hilfe ...

**Comment by Glexu** [comment]
I've been struggling with this a bit too, docker is difficult and qnap doesn't help. 

API key you need to download from the site you provided, first create an account there, then somewhere you will find it in your account. 

As for docker (you must have basic knowledge of linux):
the file for configuration is called config.json and you will find it in the /config directory in the docker instance. 
To get there, enter the container in qnap, at the top from the list choose execute -> /bin/bash and press execute

This will open a console through which you can get to the config.json file and edit it

Translated with DeepL.com (free version)

**Comment by RasputinBerry** [comment]
Hello Glexu,

Thanks for your answer. I wasn't idle and "tormented" Duckduckgo...
Yesterday I opened the docker folder in File Station with admin rights (it doesn't work without it) and found the config.json file.
Then I stopped the container in QNAP, copied out the config.json file and replaced the API key (NULL with "...key..." with Notepad++.
Then I return this copy to the original docker folder and start the container...
But: Something went wrong ... the PSACC no longer started properly and required the OPEL Login process.
To be on the safe side, I then deleted it and reinstalled it.

For me, Docker is still “a book with seven seals”.

Now I'll gather some strength and try your way...  but 2 more questions:
- Which editor can or must I call up in the terminal?
- Stop the container before or not?

Greetings

_Translated with Google_

**Comment by Glexu** [comment]
1. if you STOP the container, you can't access the files through the terminal
2. to edit files I always install via terminal MC (apt-get -y install mc) - it makes it easier to jump on files and edit them. 

I am not an expert myself, but what I described works for me without any problems

**Comment by RasputinBerry** [comment]
You are more of an expert than me....
Another stupid question:
Do I run "apt-get -y install mc" directly in the terminal of the PSACC container or do I need another container with e.g. Debian or another Linux?!

**Comment by Glexu** [comment]
you need to run it in terminal, than wait to install
when installed than type mc and hit enter
MidnightCommander starts and you can do what you want

**Comment by RasputinBerry** [comment]
... ok ... I'll try it when I have some time ... thanks so far ...

**Comment by RasputinBerry** [comment]
... unfortunately it doesn't work ... I can't install anything in the container terminal ... I'll keep looking

**Reply by Glexu** [reply]
If you need help i can try by TeamViewer

**Comment by RasputinBerry** [comment]
Here is my solution:
- log in as “admin” with WinSCP on the (my) NAS
- navigate to the file to be changed
- a double click starts the integrated editor
- Enter API
- save
- complete

(Thanks @Glexu for the first step ...)
